### PR TITLE
Add support for NamedExpr nodes

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds support for NamedExpr nodes

Unit tests are included.

One thing I'm not entirely sure of is whether a `NamedExpr`
node should be considered a user of its target.

`NamedExpr`s do evaluate to the value of the thing they are storing, so it seems
like one could make the argument that they are in fact a user of the name they
are defining.

Happy to discuss what to do about that.

### CODE REVIEW

**Comment by cpcloud** [comment]
Looking at the bytecode shows there's not actually a use of `x` and it's just store and then returning the original value after duplicating the top of the stack:

```
>>> dis.disassemble(compile(ast.parse("(x := 1)", mode="eval"), "", "eval"))
  1           0 LOAD_CONST               0 (1)
              2 DUP_TOP
              4 STORE_NAME               0 (x)
              6 RETURN_VALUE
```

File: beniget/beniget.py [hunk 1]
```diff
@@ -849,6 +849,12 @@ def visit_Subscript(self, node):
 
     visit_Starred = visit_Await
 
+    def visit_NamedExpr(self, node):
+        dnode = self.chains.setdefault(node, Def(node))
+        self.visit(node.target)
+        self.visit(node.value).add_user(dnode)
```
Comment (serge-sans-paille) [inline beniget/beniget.py:855]: I think you should visit `node.value` before `node.target` to correctly handle case like
```
a = 1
(a := a + a)
```

**Comment by serge-sans-paille** [comment]
Almost good! Just a small suggestion.

File: beniget/beniget.py (hunk 1 above)
Comment (cpcloud) [reply beniget/beniget.py:854]: Fixed!

**Comment by cpcloud** [comment]
@serge-sans-paille can you enable the workflow so we can get this merged? thanks!

**Comment by serge-sans-paille** [comment]
Back from a few days off... and this looks good now, thanks!

**Comment by cpcloud** [comment]
Sweet! Thanks for merging. Would it possible to cut a patch release?

**Pull request merged**